ledger/src/blockstore.rs

```diff
@@ -136,10 +136,27 @@ impl std::fmt::Display for InsertDataShredError {
     }
 }
 
+/// A "complete data set" is a range of [`Shred`]s that combined in sequence carry a singled
+/// serialized [`Vec<Entry>`].
+///
+/// Services such as `solana_core::window_service::WindowService` for a TVU, and
+/// `solana_core::replay_service::ReplayStage` for a TPU, drive generation of these, via direct or
+/// indirect calls to [`Blockstore::insert_shreds_handle_duplicate()`].
+///
+/// `solana_core::completed_data_sets_service::CompletedDataSetsService` is the main receiver of
+/// `CompletedDataSetInfo`.
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]
 pub struct CompletedDataSetInfo {
+    /// [`Slot`] for which we have received the last [`Shred`] in the `start_index..=end_index`
+    /// range.
     pub slot: Slot,
+
+    /// Index of the first [`Shred`] in the range of shreds that form a complete [`Vec<Entry>`].
```

Comment:
`that form a complete [`Vec<Entry>`].` -> `that belong to this set`

Reply:
The reasoning was if we say `in the range of shreds that form a complete [`Vec<Entry>`].`, the reader has to recall/make an extra mental association that this `Vec<Entry>` is referring to a completed data set. When we just say this "set" it's implied that we are talking about this exact data set represented by this struct

Reply:
Makes sense, thank you for the explanation.

I must admit that as a name “data set”, is a bit confusing to me.  A “set” makes me think about something that is unordered.  And “data” does not clarify it that much.  Except, maybe, suggesting it is something unstructured?  The only association I get, is, in a network protocol, “data” can be used to reference the effective payload.  Still, a payload, is what is commonly used.

I wonder if it would be helpful to rename data sets to something like, “entry groups”, “entry arrays”, “entry lists”, or something like that.
And then `Vec<Entry>`s can get a type with a distinctive name, like `EntryGroup`.